src/main/java/seedu/duke/parser/BookmarkParser.java

```diff
@@ -0,0 +1,126 @@
+            bookmarkAnimeCommand.setBookmarkAction(paramParts[0]);
+            break;
+        default:
+            String invalidParameter = "-" + paramGiven + " is an invalid parameter!";
+            throw new AniException(invalidParameter);
+        }
+    }
+
+    private void setFirstParameter(String paramGiven) throws AniException {
+        //Action edit(e) requires first parameter as bookmarkIndex
+        if (bookmarkAnimeCommand.getBookmarkAction().equals("e")) {
+            boolean isValidBookmarkIndex = bookmarkAnimeCommand.setBookmarkIndex(paramGiven.trim());
+            if (!isValidBookmarkIndex) {
+                String invalidBookmarkIndex = "" + paramGiven + " is an invalid parameter!"
+                        + System.lineSeparator() + " Bookmark index for edit episode requires integer.";
+                LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidBookmarkIndex);
+                throw new AniException(invalidBookmarkIndex);
+            }
+        } else {
+            boolean isEmpty = paramGiven.trim().equals("");
+            if (!isEmpty) {
+                String invalidFirstParameter = "" + paramGiven + " is an invalid parameter!"
+                        + System.lineSeparator() + " Add or Delete should not have extra param.";
+                LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidFirstParameter);
+                throw new AniException(invalidFirstParameter);
+            }
+        }
+    }
+
+    private String[] getSplitDescription(String description) throws AniException {
+        String[] paramGiven = description.split("-");
+        if (paramGiven.length > 2) {
+            String invalidDescription =  description + " has too many parameter!";
+            LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidDescription);
+            throw new AniException(invalidDescription);
+        } else if (paramGiven.length < 2) {
+            String invalidDescription =  description + " has too few parameter!";
+            LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidDescription);
+            throw new AniException(invalidDescription);
+        }
+        return paramGiven;
+    }
+
+    private void paramLengthCheck(String[] paramParts) throws AniException {
+        // Parameter Additional Field Check
+        if (paramParts.length < 2) {
+            String invalidParameter = "Parameter : " + paramParts[0] + " requires an additional field";
+            LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidParameter);
+            throw new AniException(invalidParameter);
+        } else if (paramParts.length > 2) {
+            String invalidParameter = "Parameter : " + paramParts[0] + " has too much fields";
+            LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidParameter);
+            throw new AniException(invalidParameter);
+        }
+    }
+}
```

Comment:
Would this be better if it was refactored using the utility methods available in CommandParser?

Reply:
Yea I saw, but I was considering keeping my custom messages and logging. 
I think I will take up your advice to keep things consistent.

Reply:
Thanks for the feedback.